### PR TITLE
Implement Trace Identifiers.

### PR DESCRIPTION
**Pull request by carlosalberto**
Initial implementation of Trace Identifiers.

Let me know, @pglombardo @palazzem @yurishkuro

### CODE REVIEW

File: opentracing/harness/api_check.py [hunk 1]
```diff
@@ -40,6 +40,14 @@ def check_baggage_values(self):
         """
         return True
 
+    def check_trace_identifiers(self):
+        """If true, the test will validate trace ID and :class:`Span` ID by
+        retrieving them from the trace context. If false, it will only attempt
+        to retrieve them to check the API compliance, but not actually validate
+        the values. The latter mode is only useful for no-op tracer.
```
Comment (yurishkuro) [inline opentracing/harness/api_check.py]: I am not sure what the difference between true and false. How can you **validate** the returned values? The actual content of those values is not dictated by the spec.

Comment (yurishkuro) [reply opentracing/harness/api_check.py]: ah, nvm, I see what it checks.

File: opentracing/harness/api_check.py [hunk 2]
```diff
@@ -40,6 +40,14 @@ def check_baggage_values(self):
         """
         return True
 
+    def check_trace_identifiers(self):
+        """If true, the test will validate trace ID and :class:`Span` ID by
```
Comment (yurishkuro) [inline opentracing/harness/api_check.py]: Suggest to clarify what it means by "validating", i.e. that trace IDs match on parent and child, and span IDs are not the same. Not really sure what would be the example when this does not hold, aside from situations when a traced doesn't support the new methods altogether (i.e. returns blanks).

File: opentracing/span.py [hunk 3]
```diff
@@ -30,6 +30,31 @@ class SpanContext(object):
...
+        """
+        return ''
+
+    def to_span_id(self):
+        """
+        Return the ID of the associated :class:`Span`.
+
+        Unique within a trace. Each :class:`Span` within
+        a trace contains a different ID.
+        The empty string is a valid return value, while ``None`` is not.
```
Comment (yurishkuro) [inline opentracing/span.py]: Don't recall if the spec imposes any restrictions on the allowed alphabet of the output, like "must be URL safe"?

Comment (carlosalberto) [reply opentracing/span.py]: Oh, that's a good point. I think the spec doesn't mention alphabet nor URL safety anywhere. I guess we should mention this bit there, at least ;)

File: opentracing/harness/api_check.py (hunk 2 above)
Comment (carlosalberto) [reply opentracing/harness/api_check.py]: > Not really sure what would be the example when this does not hold, aside from situations when a traced doesn't support the new methods altogether

I think this only stands for no-op `Tracer`, essentially ;)

**Comment by carlosalberto** [comment]
@yurishkuro Added an explanation about validating the trace identifiers - I *think* the url-safety note is probably more for the spec itself? Otherwise, we could merge perhaps? Let me know, etc

**Comment by carlosalberto** [comment]
(Also changed it to be merged to `v2.1.0` so we can release a RC later on)

File: opentracing/span.py (hunk 3 above)
Comment (yurishkuro) [reply opentracing/span.py]: I don't think we can require url safety in the spec, so ok

**Comment by carlosalberto** [comment]
I will merge this PR in the next days, as it now has the proper documentation, based on the Java iteration @yurishkuro @palazzem @pglombardo

**Comment by carlosalberto** [comment]
Merged! ;)

**Pull request merged**